x/retry/retry.go

```diff
@@ -163,7 +163,7 @@ func getTimeLeft(ctx context.Context, max time.Duration) (timeleft time.Duration
 func isRetryable(err error) bool {
 	// TODO(#1080) Update Error assertions to be more granular.
 	switch yarpcerrors.ErrorCode(err) {
-	case yarpcerrors.CodeInternal, yarpcerrors.CodeDeadlineExceeded:
+	case yarpcerrors.CodeInternal, yarpcerrors.CodeDeadlineExceeded, yarpcerrors.CodeUnavailable, yarpcerrors.CodeUnknown:
```

Comment:
With the policy system we do know if the request is idempotent, every one of these error codes (except for maybe unavailable) should only be used on idempotent endpoints.  This middleware is not explicitly trying to solve connection issues (that really should be done on a per transport basis).  It's just, if an unavailable error comes back for an idempotent endpoint, we might as well retry it.